auth/auth.go

```diff
@@ -237,6 +248,25 @@ func (c *Client) VerifyIDToken(idToken string) (*Token, error) {
 	return p, nil
 }
 
+// VerifyIDTokenWithCheckRevoked verifies the signature	and payload of the provided ID token and
+// if requested, whether it was revoked.
+// see: VerifyIDToken above.
+func (c *Client) VerifyIDTokenWithCheckRevoked(ctx context.Context, idToken string, checkToken bool) (*Token, error) {
```

Comment:
It seems odd to have both of these pieces of functionality in one function. `VerifyIdToken` is already exposed as an API people could use any time they would otherwise call this with `checkToken = false`. So why not remove the `checkToken` parameter and always assume it's true? Then maybe the function could be named something like `VerifyIDTokenNotRevoked`. TBH, I don't really understand why `VerifyIDToken` doesn't already check the revoked state, though.

Reply:
validSince is initialized to 0 when creating a new user, and we are only introducing revoke here. so we probably will not break anything if we assume it to be true.

Reply:
Correction, valid since is initialized to the current timestamp.